frontend/catalyst/pennylane_extensions.py

```diff
@@ -425,8 +432,13 @@ def __call__(self, *args, **kwargs):
             if argnums := self.grad_params.argnum is None:
                 argnums = 0
             if self.grad_params.scalar_out:
-                results = jax.grad(self.fn, argnums=argnums)(*args)
+                if self.grad_params.with_value:
+                    results = jax.value_and_grad(self.fn, argnums=argnums)(*args)
+                else:
+                    results = jax.grad(self.fn, argnums=argnums)(*args)
             else:
+                if with_value := self.grad_params.with_value:
+                    raise ValueError(f"with_value must be False , got {with_value}")
```

Comment:
Fairly minor, but if the user-facing API does not include a way to generate a jacobian with value instruction, then this can be an assertion instead of an exception. That will also prevent coverage issues if we don't have a test catching this exception.

Reply:
Makes sense. Done.

I have also changed to an assertion when the user tries to jitting `value_and_grad`, as it's not implemented yet. I will remove such assertion when the MLIR part is ready.